test/AutoDiff/control_flow.swift

```diff
@@ -519,4 +519,46 @@ ControlFlowTests.test("Enums") {
   }
 }
 
+ControlFlowTests.test("Loops") {
+  func for_loop(_ x: Float) -> Float {
+    var result = x
+    for _ in 1..<3 {
+      result = result * x
+    }
+    return result
+  }
+  expectEqual((8, 12), valueWithGradient(at: 2, in: for_loop))
+  expectEqual((27, 27), valueWithGradient(at: 3, in: for_loop))
+
+  func while_loop(_ x: Float) -> Float {
```

Comment:
Please also test a `repeat`-`while` loop.

Reply:
Added tests for `repeat-while` loops and `continue`/`break` statements in https://github.com/apple/swift/pull/25558/commits/32f66294d96791f67f15234948b493b76b9d31da.
Exposed [TF-584](https://bugs.swift.org/browse/TF-584): derivative computation bug for repeat-while loops. Thanks for pointing this out!